visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts

```diff
@@ -80,28 +84,27 @@ describe("screenshotButtonComponent", () => {
 		const clickDownloadLinkSpy = jest.spyOn(HTMLAnchorElement.prototype, "click").mockImplementation()
 		const makeScreenshotToFileSpy = jest.spyOn(fixture.componentInstance, "makeScreenshotToFile")
 
-		await user.click(screen.getByTitle("Take a screenshot of the map with Ctrl+Alt+S (save as file) or Ctrl+Alt+F (copy to clipboard)"))
+		await fixture.componentInstance.makeScreenshotToFile()
 
+		expect(makeScreenshotToFileSpy).toHaveBeenCalledTimes(1)
 		await waitFor(() => {
-			expect(makeScreenshotToFileSpy).toHaveBeenCalledTimes(1)
 			expect(clickDownloadLinkSpy).toHaveBeenCalledTimes(1)
 		})
 	})
 
 	it("should be disabled when write to clipoard is not available in browser", async () => {
-		const user = userEvent.setup()
+		;(checkWriteToClipboardAllowed as jest.Mock).mockImplementation(() => false)
 		const { container, fixture } = await render(ScreenshotButtonComponent, {
 			excludeComponentDeclaration: true
 		})
 
 		const makeScreenshotToClipboardSpy = jest.spyOn(fixture.componentInstance, "makeScreenshotToClipboard")
 
-		await user.click(screen.getByTitle("Firefox does not support copying to clipboard"))
+		await fixture.componentInstance.makeScreenshotToClipboard()
```

Comment:
How do we signal to the user when copying to clipboard is not supported? Previously we tested whether or not a, i think, popup (?) was displayed but now we only test if the function was called. So how do we inform the user of being unable to copy to clipboard and should we test this method too?

Reply:
In Firefox the screenshot button is disabled. This is still tested in line 103.